custom_components/xiaomi_miio/coordinator.py

```diff
@@ -30,43 +35,53 @@ def __init__(self, hass: HomeAssistant, device: Device) -> None:
         )
         self._device = device
 
-    # TODO: cleanup async_update_data() to allow tries to avoid code duplication
+    def __defer_or_raise(self, ex: Exception):
+        self.retry_count -= 1
+        if self.retry_count <= 0:
+            # Passtru exception to hass
+            raise ex
+        return self.saved_state
```

Comment:
It's probably a good idea to add at least a debug log message whenever we are returning information from cache?